pkg/controller/status.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"time"
+
+	controllererrors "github.com/gardener/gardener/extensions/pkg/controller/error"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+
+	ciliumv1alpha1 "github.com/gardener/gardener-extension-networking-cilium/pkg/apis/cilium/v1alpha1"
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func (a *actuator) updateProviderStatus(
+	ctx context.Context,
+	network *extensionsv1alpha1.Network,
+	config *ciliumv1alpha1.NetworkConfig,
+) error {
+	status, err := a.ComputeNetworkStatus(config)
+	if err != nil {
+		return err
+	}
+
+	network.Status.ProviderStatus = &runtime.RawExtension{Object: status}
+	network.Status.LastOperation = extensionscontroller.LastOperation(gardencorev1beta1.LastOperationTypeReconcile,
+		gardencorev1beta1.LastOperationStateSucceeded,
+		100,
+		"Cilium was configured successfully")
```

Comment:
`extensionscontroller.LastOperation` always sets `LastUpdateTime` to `time.Now` regardless of any changes of the state, description or progress https://github.com/gardener/gardener-extension-networking-cilium/blob/c042dc0f4d90697db9a8838a05328d8aec28f0ef/vendor/github.com/gardener/gardener/extensions/pkg/controller/status.go#L25-L26

Call update only when there is an actual change between the `desired` and `actual` status of `LastOperation`.

Reply:
https://github.com/gardener/gardener-extension-networking-cilium/pull/7#discussion_r393890644